packages/common/utils/src/object/split.ts

```diff
@@ -0,0 +1,9 @@
+/** @tossdocs-ignore */
+import { ObjectKeys, omit, pick } from '.';
+
+export function split<ObjectType extends Record<PropertyKey, unknown>, KeyTypes extends Array<ObjectKeys<ObjectType>>>(
+  obj: ObjectType,
+  keys: KeyTypes
+): [ReturnType<typeof pick<ObjectType, KeyTypes>>, ReturnType<typeof omit<ObjectType, KeyTypes>>] {
+  return [pick(obj, keys), omit(obj, keys)];
```

Comment:
This would be more efficient. This iterates over all keys in `obj` only once.

```typescript
const keysSet = new Set(keys);
const picked = {};
const omitted = {};

for (const key in obj) {
  if (keysSet.has(key)) {
    picked[key] = obj[key];
  } else {
    omitted[key] = obj[key];
  }
}

return [picked, omitted];
```

Reply:
thank you!